src/transformers/pipelines/image_to_text.py

```diff
@@ -78,29 +84,43 @@ def __call__(self, images: Union[str, List[str], "Image.Image", List["Image.Imag
         Args:
             images (`str`, `List[str]`, `PIL.Image` or `List[PIL.Image]`):
                 The pipeline handles three types of images:
-
                 - A string containing a HTTP(s) link pointing to an image
                 - A string containing a local path to an image
                 - An image loaded in PIL directly
-
                 The pipeline accepts either a single image or a batch of images.
-
             max_new_tokens (`int`, *optional*):
                 The amount of maximum tokens to generate. By default it will use `generate` default.
-
             generate_kwargs (`Dict`, *optional*):
                 Pass it to send all of these arguments directly to `generate` allowing full control of this function.
-
         Return:
             A list or a list of list of `dict`: Each result comes as a dictionary with the following key:
-
             - **generated_text** (`str`) -- The generated text.
         """
         return super().__call__(images, **kwargs)
 
-    def preprocess(self, image):
-        image = load_image(image)
-        model_inputs = self.image_processor(images=image, return_tensors=self.framework)
+    def preprocess(self, images, preprompt=None):
+        images = load_image(images)
+
+        if preprompt is not None and not isinstance(preprompt, str):
+            raise ValueError(
+                f"Received an invalid text input, got - {type(preprompt)} - but expected a single string. "
+                "Note also that one single text can be provided for conditional image to text generation."
+            )
+
+        # check if the model is not a pix2struct model
+        if preprompt is not None and self.model.config.model_type == "pix2struct":
+            model_inputs = self.image_processor(images=images, header_text=preprompt, return_tensors=self.framework)
```

Comment:
Shouldn't we add this logic to the VQA pipeline? Rendering text on images is only done for VQA right?